src/azure-cli/azure/cli/command_modules/hdinsight/_params.py

```diff
@@ -255,3 +277,58 @@ def load_arguments(self, _):
                        help='The name of the cluster.')
             c.argument('hosts', options_list=['--host-names'], nargs='+',
                        help='A space-delimited list of host names that need to be restarted.')
+
+        # autoscale
+        autoscale_commands = ['create', 'update', 'delete', 'show']
+        for command in autoscale_commands:
+            with self.argument_context('hdinsight autoscale ' + command) as c:
+                c.argument('cluster_name', options_list=['--cluster-name'],
+                           completer=get_resource_name_completion_list('Microsoft.HDInsight/clusters'),
+                           help='The name of the cluster.')
+
+        for command in ['create', 'update']:
+            with self.argument_context('hdinsight autoscale ' + command) as c:
+                c.argument('min_workernode_count', arg_group='Load-based Autoscale',
+                           help='The minimal workernode count for Load-based atuoscale.')
+                c.argument('max_workernode_count', arg_group='Load-based Autoscale',
+                           help='The max workernode count for Load-based atuoscale.')
+                c.argument('timezone', arg_group='Schedule-based Autoscale', validator=validate_timezone_name,
+                           completer=get_generic_completion_list(AUTOSCALE_TIMEZONES),
+                           help='The timezone for schedule autoscale type. '
+                                'Values from `az hdinsight autoscale list-timezones`')
+
+        with self.argument_context('hdinsight autoscale create') as c:
+            c.argument('type', arg_type=get_enum_type(known_autoscale_types), help='The autoscale type.')
+            c.argument('days', nargs='+', arg_group='Schedule-based Autoscale',
+                       completer=get_generic_completion_list(week_days),
+                       help='A space-delimited list of schedule day. Valid days are {}.'.format(', '.join(week_days)))
+            c.argument('time', arg_group='Schedule-based Autoscale', validator=validate_time,
+                       help='The 24-hour time in the form xx:xx in days.')
+            c.argument('workernode_count', options_list=['--workernode-count'], arg_group='Schedule-based Autoscale',
```

Comment:
>workernode_count [](start = 24, length = 16)

add type=int

Reply:
fixed

---
In reply to: [468398033](https://github.com/Azure/azure-cli/pull/14692#discussion_r468398033) [](ancestors = 468398033)